include/deal.II/matrix_free/fe_evaluation_base_data.h

```diff
@@ -136,6 +152,23 @@ class FEEvaluationBaseData
   FEEvaluationBaseData &
   operator=(const FEEvaluationBaseData &other);
 
+  /**
+   * Sets the pointers for values, gradients, hessians to the central
+   * scratch_data_array inside the given scratch array, for a given number of
+   * components as provided by one of the derived classes.
+   */
+  void
+  set_data_pointers(AlignedVector<VectorizedArrayType> *scratch_data,
+                    const unsigned int                  n_components);
```

Comment:
Could we extend this for the case that we have different polynomial degrees/number of quadrature points as in the case of hp and simplex meshes?

Reply:
I don't understand this question. For different degrees we would have different `FEEvaluation` objects?

Reply:
The question was targeted towards what is missing to be able to deal with a single `FEEvaluation` different degrees.

Reply:
To support different degrees within the same `FEEvaluation` object, there is quite some additional work to be done, because we would need to change the `shape_info` pointers, several `const` member variants (number of q points, active fe index, etc) at `reinit`. It would basically boil down to constructing a new `FEEvaluation` object as we do it now, it's only the `mapping_info` and `dof_info` data structure that would remain alive.